src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -4,45 +4,80 @@
 
 import java.util.List;
 
-import seedu.address.commons.core.index.Index;
+import seedu.address.commons.core.index.Indices;
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.person.Person;
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
+ * Deletes one or more fosterers identified using their displayed index from the address book.
  */
 public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
+            + ": Deletes fosterers identified by the index number used in the displayed person list.\n"
             + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + "Example: " + COMMAND_WORD + " 1 2 3";
 
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Fosterer: %1$s";
 
-    private final Index targetIndex;
+    public static final String MESSAGE_DELETE_PEOPLE_SUCCESS = "Deleted Fosterers: %1$s";
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    private final Indices targetIndices;
+
+    public DeleteCommand(Indices targetIndices) {
+        this.targetIndices = targetIndices;
+    }
+
+    /**
+     * Checks if zero-based target index is valid.
+     */
+    public static boolean isValidIndex(int targetIndex, int listSize) throws CommandException {
+        if (targetIndex >= listSize || targetIndex < 0) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+        return true;
+    }
+
+    /**
+     * Returns the people to delete, given the last shown list of Persons.
+     *
+     * @throws CommandException when target index is out of bounds.
+     */
+    public Person[] getPeopleToDelete(List<Person> lastShownList) throws CommandException {
+        int[] zeroBasedIndices = this.targetIndices.getZeroBased();
+        int numberOfDeletions = this.targetIndices.getSize();
+        Person[] peopleToDelete = new Person[numberOfDeletions];
+
+        for (int i = 0; i < numberOfDeletions; i++) {
+            int targetIndex = zeroBasedIndices[i];
+
+            if (isValidIndex(targetIndex, lastShownList.size())) {
+                Person personToDelete = lastShownList.get(targetIndex);
+                peopleToDelete[i] = personToDelete;
+            }
+        }
```

Comment:
nit: stream for higher level of abstraction
```suggestion
        int[] zeroBasedIndices = this.targetIndices.getZeroBased();
        return Arrays.stream(zeroBasedIndices)
                .mapToObj(this::getPersonIfValidIndex)
                .filter(Objects::nonNull)
                .toArray(Person[]::new);
```
optional suggestion.

Reply:
Tried this but I will have to handle an exception within the mapper due to my function throwing a command exception :P hence I'm keeping the current implementation, thanks though!